### PR TITLE
Replace HttpBin.org/response-headers Tests with WebListener

### PR DESCRIPTION
**Pull request by markekraus**
* Adds `/ResponseHeaders/` Test to WebListener
* Adds `-Query` parameter to `Get-WebListenerUrl` which accepts a dictionary where keys are wuery string fields and values are query string values
* Replaces http://httpbin.org/response-headers tests with WebListener

Reference #2504 

(note: the tests as they were intended before were actually not accurate. http://httpbin.org/response-headers returns an `Content-Type: application/json` in addition to `Content-Type: ` from the supplied header. The implementation in this PR does allow for a blank/missing `Content-Type` to be returned. It is possible this may reveal errors that were not visible before.)

### CODE REVIEW

File: .spelling
Comment (iSazonov) [inline .spelling]: Are they really all used?

File: test/tools/WebListener/Controllers/ResponseHeadersController.cs
Comment (iSazonov) [inline test/tools/WebListener/Controllers/ResponseHeadersController.cs]: Maybe add newline.

**Review by iSazonov — APPROVED** [review]
LGTM with two minor comments.

File: .spelling
Comment (markekraus) [reply .spelling]: I'm not sure what you are asking. But, I added a file level override because I had been putting all of the overrides for this in the Global dictionary.  That was probably a poor decision, so I'm adding the new section and will do a separate PR after this is merged to move the ones from my previous PRs from global to the file override.

File: test/tools/WebListener/Controllers/ResponseHeadersController.cs
Comment (markekraus) [reply test/tools/WebListener/Controllers/ResponseHeadersController.cs]: fixed.

File: .spelling
Comment (iSazonov) [reply .spelling]: Thanks. Clear.

Closed.

**Comment by markekraus** [comment]
Travis CI fail due to #5062

**Comment by iSazonov** [comment]
@markekraus MSFT maintainers address such issues very fast. Be patient :-)

**Comment by markekraus** [comment]
@iSazonov they do :smile:. But the community can also address them if an issue is filed. Besides, I prefer to have an issue that explains why my PRs are failing externally. Makes it easier to track when the external issue has been resolved to move forward. @SteveL-MSFT would likele create an issue for this when he came online anyway. I'm no no hurry, just trying to be helpful and thorough.

**Comment by SteveL-MSFT** [comment]
@markekraus looking at the Mac issue, restarted your build for now

**Comment by markekraus** [comment]
@TravisEz13 Is there something outstanding on this PR I need to do?

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**
